app/decorators/category_decorator.rb

```diff
@@ -0,0 +1,11 @@
+class CategoryDecorator < Draper::Decorator
+  delegate_all
+
+  def self.collection_decorator_class
+    PaginatingDecorator
+  end
+
+  def full_amount
+    self_and_descendents.map(&:amount).flatten.reduce(0, :+)
```

Comment:
@sinpai let's make service out of `full_amount` and move to model:
```ruby
def full_amount
  Categories::Amount.new(self).calculate
end
```

this service have to encapsulate logic from the model:
```ruby
def descendents
  sub_categories.map do |sub_category|
    [sub_category] + sub_category.sub_categories + sub_category.descendents
  end.flatten.uniq
end

def self_and_descendents
  [self] + descendents
end
```